pkg/optimization/substats.go

```diff
@@ -0,0 +1,92 @@
+package optimization
+
+import (
+	"fmt"
+	"github.com/genshinsim/gcsim/internal/simulator"
+	"github.com/genshinsim/gcsim/pkg/core"
+	"go.uber.org/zap"
+)
+
+type SubstatOptimizer struct {
+	logger     *zap.SugaredLogger
+	optionsMap map[string]float64
+	Details    *SubstatOptimizerDetails
```

Comment:
i don't think it's necessary to have a pointer reference here as the methods are already pointer methods